spec/CloudCode.spec.js

```diff
@@ -2451,3 +2451,86 @@ describe('beforeLogin hook', () => {
     done();
   });
 });
+describe('Retrieve subclass objects', () => {
+  it('Retrieve objects with no inited properties', async done => {
+    class SubclassObject extends Parse.Object {
+      constructor(attributes) {
+        super('SubclassObject', attributes);
+      }
+      get name() {
+        return this.get('name');
+      }
+      set name(val) {
+        this.set('name', val);
+      }
+    }
+
+    Parse.Object.registerSubclass('SubclassObject', SubclassObject);
+
+    Parse.Cloud.define('SubclassObject:first', () => {
+      return new Parse.Query(SubclassObject).first();
+    });
+
+    await new SubclassObject({ name: 'Foo' }).save();
+    const result = await Parse.Cloud.run('SubclassObject:first');
+
+    expect(result instanceof SubclassObject).toBeTruthy();
+    expect(result.name).toBe('Foo');
+    done();
+  });
+
+  it('Retrieve objects with simple inited properties', async done => {
+    class SubclassSimpleProp extends Parse.Object {
+      constructor(attributes) {
+        super('SubclassSimpleProp', attributes);
+        this.name = 'Initial';
+      }
+      get name() {
+        return this.get('name');
+      }
+      set name(val) {
+        this.set('name', val);
+      }
+    }
+
+    Parse.Object.registerSubclass('SubclassSimpleProp', SubclassSimpleProp);
+
+    Parse.Cloud.define('SubclassSimpleProp:first', () => {
+      return new Parse.Query(SubclassSimpleProp).first();
+    });
+
+    await new SubclassSimpleProp({ name: 'Foo' }).save();
+    const result = await Parse.Cloud.run('SubclassSimpleProp:first');
+
+    expect(result instanceof SubclassSimpleProp).toBeTruthy();
+    expect(result.name).toBe('Foo');
+    done();
+  });
+
+  it('Retrieve objects with inited array properties', async done => {
+    class SubclassArrayProp extends Parse.Object {
+      constructor(attributes) {
+        super('SubclassArrayProp', attributes);
+        this.myArray = ['some', 'initial', 'values'];
+      }
+      get myArray() {
+        return this.get('myArray');
+      }
+      set myArray(val) {
+        this.set('myArray', val);
+      }
+    }
+
+    Parse.Object.registerSubclass('SubclassArrayProp', SubclassArrayProp);
+    Parse.Cloud.define('SubclassArrayProp:first', () => {
+      return new Parse.Query(SubclassArrayProp).first();
+    });
+
+    await new SubclassArrayProp({ myArray: ['custom', 'value'] }).save();
```

Comment:
This save call doesn't work. Your super constructor will set the attributes. The very next line `this.myArray = ...` overrides those attributes.

A workaround would be to do the following or something similar
```
constructor(attributes) {
  super('SubclassArrayProp', attributes);
  if (!attributes.myArray) {
    this.myArray = ['some', 'initial', 'values'];
  }
}
```

Reply:
I tested your suggested workaround in this [commit](https://github.com/RaschidJFR/Parse-SDK-JS/commit/d0f48aa0dc4eeab1be54aafc501b1ba1cbeaface) but the case is still failing. For a moment I started thinking it was a mere coding logic issue, but now I see it is not.

I'll dig in a bit more as soon as I get a chance.